### DISCUSSION TITLE
Firewall Ports

### DISCUSSION THREAD

**Discussion by istenburg** [streamlink/streamlink-twitch-gui#998] [General]
Which ports do I have to release in the firewall so that everything works?

**Comment by bastimeyer** [comment]
There's only one relevant for Streamlink Twitch GUI. In order for applications to be able to access the Twitch API, users have to authenticate. The authentication response from your web browser is sent to [a locally run HTTP server by Streamlink Twitch GUI, which is listening on TCP port 65432](https://github.com/streamlink/streamlink-twitch-gui/blob/v2.4.1/src/config/twitch.json#L13-L27).

**Comment by istenburg** [comment]
If I enable http and https in the firewall, everything is loaded except the thumbnails...

**Reply by bastimeyer** [reply]
This sounds like a DNS issue. I can't help you fix your network configuration. You'll have to do this yourself. You didn't even mention which OS you're using...

Streamlink Twitch GUI doesn't do anything special. It simply displays the images with the URLs from the Twitch API data. Thumbnail images are usually hosted on `static-cdn.jtvnw.net`, so check if this hostname properly resolves on your system. Good luck.

**Comment by istenburg** [comment]
I use Arch Linux, sorry...
If I close the firewall in both directions, nothing works at first. Then I open individual ports, like 80, 443 etc. Then the channels are loaded, but no preview images. Only when I open the firewall again after going out does everything work again. So I thought that Streamlink Twitch Gui uses special ports!
